scripts/no-now-lint.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+# This script looks for all uses of a time.Now() function.
+# If any improper uses are found, they will be outputted to stderr and the script will exit with code 1.
+# If there isn't anything of concern, nothing will be outputted, and the script will exit with code 0.
+# Providing the -v or --verbose flag (or exporting VERBOSE=1) will make this output middle-step information.
+
+if [ "$1" == '-v' ] || [ "$1" == '--verbose' ]; then
+    VERBOSE='1'
+fi
+
+# Find all files that import the "time" module, and identify what they're referenced as in that file.
+# There's at least one other module that has a .Now() function ("github.com/tendermint/tendermint/types/time") that we don't want used.
+# So we're actually finding an import of any "time" package.
+# If a file has multiple such imports, there'll be a line for each import.
+# The vendor directory is ignored since it's not under our control and has lots of matches (that hopefully don't cause problems).
+# Output is in the format of "<file>:<alias>", e.g. "app/test_helpers.go:time".
+#
+# First, find all go files of possible interest.
+# Then, grep each go file looking for an import of a "time" package.
+# Transform each entry into the format "<file>:<alias>" by:
+#  1. Changing ': <alias> "time"' or ':import <alias> "time"' into ':<alias>'.
+#  2. Changing ': "time"' or ':import "time"' into ':time'.
+#  3. Removing the leading './' on the filename.
+# Then sort them because I'm pedantic like that.
+time_imports="$( \
+    find . -type f -name '*.go' -not -path '*/vendor/*' \
+    | xargs grep -E '^(import)?[[:space:]]+([[:alnum:]]+[[:space:]]+)?"([^"]+/)?time"' \
+    | sed -E -e 's/:(import)?[[:space:]]+([[:alnum:]]+)[[:space:]]+"([^"]+\/)?time".*$/:\2/' \
+             -e 's/:(import)?[[:space:]]+"([^"]+\/)?time".*$/:time/' \
+             -e 's/^\.\///' \
+    | sort
+)"
+[ -n "$VERBOSE" ] && printf 'Time imports:\n%s\n\n' "$time_imports"
+
+# Find all uses of a time.Now() function.
+# Loop through each line of time imports.
+# Split the line into the file and the alias.
+# Grep the file for <alias>.Now() taking care to not include entries of a different alias that ends the same.
+now_uses="$( \
```

Comment:
This is an extremely rare and stupid edge case, but someone may include time.Now() inside a comment. "// Using xyz because time.Now() is banned" or if they simply comment it out "// time.Now()".

Reply:
Good call. I've updated the script to ignore such "matches."